src/ffi/spatialmath/axis_angle.rs

```diff
@@ -0,0 +1,69 @@
+use ffi_helpers::null_pointer_check;
+use nalgebra::{Quaternion, UnitQuaternion};
+
+use crate::spatialmath::utils::AxisAngle;
+
+/// Allocates a copy of the axis angle to the heap with a stable memory address and
+/// returns the raw pointer (for use by the FFI interface)
+fn to_raw_pointer(aa: &AxisAngle) -> *mut AxisAngle {
+    Box::into_raw(Box::new(*aa))
+}
+
+/// Free memory at the address of the axis angle pointer. Outer processes
+/// that work with axis angles via the FFI interface MUST remember 
+/// to call this function when finished with an axis angle instance
+/// 
+/// # Safety
```

Comment:
do we need an actual safety comment here? Should the line about remembering to call this be in the Safety section?

Reply:
I'll put it in both places

Reply:
oh I misunderstood what you meant, yeah will change